README.md

```diff
@@ -1,32 +1,254 @@
-- The Google API Code Synchronizer, which can merge updates in the generated
-  code wrapper to an existing code wrapper which might have manual edits.
+- Java
+- Python
+- Go
+- C#
+- Ruby
+- Node.js
+- PHP
 
-More Google API tools will be introduced in the future.
+## Usage
 
-## The Google API Code Generator
+There are two main ways to generate clients:
 
-We currently support the generation of client code wrappers in seven languages:
-Java, Python, Go, C#, Ruby, Node.js, and PHP. There are two ways to generate client
-code wrappers:
+1. Build from source and run directly. This is detailed below.
+2. Invoke a code generation pipeline through
+   [artman]( https://github.com/googleapis/artman/#usage). This takes care of several of the steps
+   below and is not documented here.
 
-1. Invoking directly through a gradle task (see the build.gradle:runCodeGen task).
-2. Invoking a code generation pipeline through
-   [artman]( https://github.com/googleapis/artman/#usage).
+### Prerequisites for code generation
 
-## The Google API Code Synchronizer
+1. You need a proto file describing your API. Proto files of existing Google APIs are available at
+[googleapis](https://github.com/googleapis/googleapis).
+  - Example: https://github.com/googleapis/googleapis/blob/master/google/cloud/language/v1/language_service.proto
+2. You need a yaml file describing certain service values. This is unfortunately not
+well-documented.
+  - Example: https://github.com/googleapis/googleapis/blob/master/google/cloud/language/language_v1.yaml
+3. Java 8 JDK
 
-The process currently hard-wires `kdiff3` for merging and conflict resolution.
-`kdiff3` runs in auto mode and will only pop up a UI if conflict resolution is
-not possible (note: X display needed in this case). There are two ways to invoke
-code synchronizer:
+### Process overview for generating a client library
 
-1. Invoking directly through a gradle task (see the build.gradle:runSynchronizer
-   task).
-2. Invoking a code generation pipeline which contains synchronization step
-   through [artman](https://github.com/googleapis/artman/#usage).
+1. Generate a descriptor file from the proto (once per API)
+2. Run client config generation (once per API)
+3. Manually tweak the generated client config
+4. Run code generation (once per API + language)
+
+Each of these steps are described in more detail below.
+
+### Set up prerequisites for building from source and running directly
+
+First, you need to install protoc if you don't have it yet: see
+[protocol-compiler-installation](https://github.com/google/protobuf#protocol-compiler-installation).
+The pre-built binaries are the easiest, but there aren't really instructions for installation -
+you have to just download the correct binaries for your platform at put them on your path.
+If you are using Linux x86_64:
+
+1. Download the latest zip from https://github.com/google/protobuf/releases ending in "-linux-x86_64.zip"
```

Comment:
Note: we don't require a specific version of protobuf. Fairly old versions can still be used, actually. I don't want a direct link because I don't want to imply that's the only platform that will work, and I also want people to prefer newer versions when protobuf releases them. This readme isn't specific to linux - it works on Mac too (in fact, so far I have only verified on Mac). 

That's a fair point about supported platforms - I'll add a note.

Reply:
PHP requires at least 3.5.1 I think (because of various bugs in PHP support in protobuf)

Reply:
Oh ok. Well anyway, the instructions are correct for that (since they say "latest zip").